### ISSUE TITLE
[Class][Abilities] Death Knight runes

### ISSUE THREAD

**Issue by MrAndersonX** [dalaranwow/dalaran-wow#3921]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
 Alot of times lately, I've been getting a weird bug/glitch concerning runes. I don't know if it's caused by Death Runes, or if it's a visual glitch, but ALOT of times during quests/raids/dungeons etc when I use my skills by using runes, sometimes I can have a unholy and frost rune up but I can't use a skill that consumes them, while getting the error message "That ability is not ready yet". Even tho I have the req runes available, I cannot use that ability until the next unholy/frost runes get off cd. I literally have no idea what's the cause of this problem, and will try to get more ps as evidence of it. Problem is, it's kind of hard to do cause it only occurs sometimes and I have no idea what's triggering it.
 
**Expected behaviour**: Tell us what should happen instead
 If you have available runes, you should be able to use your abilities (Death Strike, Scourge Strike, Heart Strike etc) without getting a error message and without waiting for the other runes to be off cd.

**Steps to reproduce the problem**:

1.  Use the abilities.
2.  Get runes on cd.
3.  Try to use the abilities again.

**Include proofs for this behaviour**

 Here I had to use Blood Strike because it did not let me use Scourge Strike, telling me that "ability is not ready yet", even though as you can see I had 1 unholy and frost up, and the other 2 had to finish their cd before I could use it. Extremly annoying bug.  Will try to get more proof when I encounter it again.
 https://postimg.org/image/6hezt5ykl/
  

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Rushor** [comment]
frist fix for this: https://github.com/TrinityCore/TrinityCore/commit/94bb71e5a1f2b341e94aa6adc661f1200b2b10aa (already on ptr)

second fix will follow soon.

**Comment by MrAndersonX** [comment]
Thanks ! Thought I was being paranoid at a point, but after it happened to me several times I knew smth was wrong :).

**Comment by Rushor** [comment]
More information for the reason of the issue:

If you use spell=45529 your are sometimes not able to use the Death Runes. Other reason for bugs with runes i could not find/reproduce so far.

**Comment by Rushor** [comment]
frist fix for this: https://github.com/TrinityCore/TrinityCore/commit/94bb71e5a1f2b341e94aa6adc661f1200b2b10aa (already on ptr)
https://github.com/TrinityCore/TrinityCore/commit/868dcb001cd5116f2a30ecfef6a1ba888d449abe (already on ptr)

Scripts/Spells: Death Knight runes should always be usable after use of "Blood Tap"

* closes https://github.com/dalaranwow/dalaran-wow/issues/3921
* commit: a35a2d954cf5f780f14d6a16ae2ec168b9046c63

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by robinsch** [comment]
@Rushor I ported this to PTR.

**Comment by TeridanDW** [comment]
Hello, this issue is pretty hard to reproduce and after a lot of attempts I couldn't, so I will consider it as fixed, but I'll be happy if our new team member can verify me here.

**Comment by AttendantTock** [comment]
Hello,

I think it happens when you spam something and miss it, the runes cooldown are not showing properly.
In my exemples, I am spamming Scourge Strike (1 Frost, 1 Unholy) and Blood Strike to turn Blood into Death.

This first (bad) screenshots is taken after a SS miss, and we can see only one Frost strike on cooldown.
![capture d ecran 2017-10-22 a 22 36 18](https://user-images.githubusercontent.com/29557022/31866740-0dd10536-b77c-11e7-92b6-e0a61902f825.png)

This second screenshot is taken in better conditions. We can see, in 5 seconds, 1 SS miss and two SS hits. Or, only two runes are in cooldown and 4 are shown as available. It complains about not having enough runes however.
![capture d ecran 2017-10-22 a 22 46 39](https://user-images.githubusercontent.com/29557022/31866765-6114f342-b77c-11e7-84a1-fe949f3d542f.png)

I think the runes availability is correct, and it's just a graphic update issue when skills are spammed and missed. (?)
